soroban-sdk/src/crypto/bls12_381.rs

```diff
@@ -101,6 +152,57 @@ impl_bytesn_repr!(G2Affine, 192);
 impl_bytesn_repr!(Fp, 48);
 impl_bytesn_repr!(Fp2, 96);
 
+impl Fp {
+    pub fn env(&self) -> &Env {
+        self.0.env()
+    }
+
+    pub fn map_to_g1(&self) -> G1Affine {
```

Comment:
It looks like this method converts an `Fp` to a `G1Affine`. Is that accurate? Or is there semantics about the `map` part of it that represent more than a conversion from one type to the next?

Rust naming conventions suggest using `into`, `as_`, and `to_` for type conversions, so it'd be preferred to follow that naming convention unless this shouldn't be thought of as a conversion. The most appropriate for this one is probably `to_g1` because it's a borrowed to owned. See https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv.

```suggestion
    pub fn to_g1(&self) -> G1Affine {
```

Reply:
actually this is not a simple conversion, but performs mapping algorithms to map a number into a point. (the particular method is described [here](https://www.rfc-editor.org/rfc/rfc9380.html#name-simplified-swu-for-ab-0)). I've also added a comment in the code.